src/query/service/src/pipelines/processors/transforms/hash_join/transform_hash_join_build.rs

```diff
@@ -102,13 +104,14 @@ impl TransformHashJoinBuild {
     // Called after processor read spilled data
     // It means next round build will start, need to reset some variables.
     fn reset(&mut self) -> Result<()> {
-        let mut count = self.build_state.row_space_builders.lock();
-        *count += 1;
-        let mut count = self.build_state.hash_join_state.hash_table_builders.lock();
         self.finalize_finished = false;
-        *count += 1;
         // Only need to reset the following variables once
-        if *count == 1 {
+        if *self.build_state.row_space_builders.lock() == 0 {
```

Comment:
There may be two threads enter this statement block at the same time and get the same build_worker_num.